File: src/main/java/cc/mirukuneko/realtrainmodrenewed/util/UnitConverter.java
package cc.mirukuneko.realtrainmodrenewed.util;

import static cc.mirukuneko.realtrainmodrenewed.util.RealTrainModRenewedConstants.TICK_PER_SECOND;

/// 単位変換メソッド
public final class UnitConverter {
    private UnitConverter() {
    }

    /// 速度の単位キロメートル毎時をブロック毎ティックに変換する
    public static float kph2bpt(float kilometrePerHour) {
        return kilometrePerHour / 3.6f / TICK_PER_SECOND;
    }

    /// 距離の単位センチメートルをメートルに変換する
    public static float cm2m(float centimetre) {
        return centimetre / 100;
    }

    /// 時間の単位秒をティックに変換する
    public static float s2t(float second) {
        return second * TICK_PER_SECOND;
    }

    /// 加速度の単位メートル毎秒毎秒をブロック毎ティック毎ティックに変換する
    public static float mpss2bpts(float meterPerSecondSquared) {
        return meterPerSecondSquared / (TICK_PER_SECOND * TICK_PER_SECOND);
    }

}

File: src/main/kotlin/cc/mirukuneko/realtrainmodrenewed/util/UnitConverter.kt
// SPDX-License-Identifier: LGPL-3.0-or-later
// Copyright © 2026 mirukuneko and RealTrainModRenewed contributors
package cc.mirukuneko.realtrainmodrenewed.util

import kotlin.math.roundToInt as kotlinRoundToInt

object UnitConverter {
    @JvmStatic
    fun cm2m(cm: Float): Float = cm / 100.0f

    @JvmStatic
    fun kph2bpt(kph: Float): Float = (kph * 1000.0f / 3600.0f) / 20.0f

    @JvmStatic
    fun s2t(seconds: Float): Float = seconds * 20.0f

    @JvmStatic
    fun mpss2bpts(mpss: Float): Float = mpss / (20.0f * 20.0f)
}

fun cm2m(cm: Float): Float = UnitConverter.cm2m(cm)
fun kph2bpt(kph: Float): Float = UnitConverter.kph2bpt(kph)
fun s2t(seconds: Float): Float = UnitConverter.s2t(seconds)
fun mpss2bpts(mpss: Float): Float = UnitConverter.mpss2bpts(mpss)
